app.rb

```diff
@@ -0,0 +1,128 @@
+require_relative './book'
+require_relative './student'
+require_relative './classroom'
+require_relative './teacher'
+require_relative './rental'
+
+class App
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  def list_of_all_books
+    if @books.empty?
+      puts 'You have no recorded books yet!'
+    else
+      @books.each do |book|
+        puts "Title: \"#{book.title}\", Author: #{book.author}"
+      end
+      puts
+    end
+  end
+
+  def list_of_all_people
+    if @persons.empty?
+      puts 'You have no recorded people yet!'
+    else
+      @persons.each do |person|
+        category = person.is_a?(Student) ? '[Student]' : '[Teacher]'
+        puts "#{category} Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+      puts
+    end
+  end
+
+  def create_book
+    print 'Enter Book Title: '
+    title = gets.chomp
+    print 'Enter Book Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    # push new book to the end of the array
+    @books << book
+    puts 'Book is registered!'
+  end
+
+  def create_student
+    print 'Enter Age: '
+    age = gets.chomp
+    print 'Enter Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]:'
+    permission = gets.chomp
+    room = ClassRoom.new('A')
+    student = Student.new(room, age, name, parent_permission: permission)
+    # push new book to the end of the array
+    @persons << student
+    puts 'Registration success!'
+  end
+
+  def create_teacher
+    print 'Enter Age: '
+    age = gets.chomp
+    print 'Enter Name: '
+    name = gets.chomp
+    print 'Enter Specialization: '
+    special = gets.chomp
+    teacher = Teacher.new(special, age, name)
+    # push new book to the end of the array
+    @persons << teacher
+    puts 'Registration success!'
+  end
```

Comment:
- [ ] I still can't create a student or teacher because of `parent_permission` please make sure your app working properly before submitting again  in order to complete the requirements
<img width="1078" alt="Screenshot 2023-01-07 at 11 36 59" src="https://user-images.githubusercontent.com/61790587/211146189-bde32f79-15e2-4d1c-81e1-a4577fb840e8.png">